src/libraries/System.Private.Xml/src/System/Xml/Core/XmlSubtreeReader.cs

```diff
@@ -1513,7 +1513,7 @@ private bool InNamespaceActiveState
 
         private void SetEmptyNode()
         {
-            Debug.Assert(_tmpNode.localName == string.Empty && _tmpNode.prefix == string.Empty && _tmpNode.name == string.Empty && _tmpNode.namespaceUri == string.Empty);
+            Debug.Assert(_tmpNode.localName.Length == 0 && _tmpNode.prefix.Length == 0 && _tmpNode.name.Length == 0 && _tmpNode.namespaceUri.Length == 0);
```

Comment:
This should be reverted.  A tiny perf difference doesn't matter in the context of an assert, and if one of these were to be null, we'd want to get the assert about the failure rather than a null ref.

Reply:
```suggestion
            Debug.Assert(_tmpNode.localName== string.Empty && _tmpNode.prefix == string.Empty && _tmpNode.name == string.Empty && _tmpNode.namespaceUri == string.Empty);
```